site/content/docs/custom-layout.md

```diff
@@ -0,0 +1,149 @@
+---
+title: How to create custom page types and layouts
+description: You may have many different types of notes that you want to display in different ways on your Flowershow website. In this tutorial we will cover how to add a new `Recipe` document type for our cooking blog as well as a custom `RecipeLayout` for that specific document type.
+type: Blog
+created: 2023-03-02
+authors: [philippe-du-preez]
+---
+
+You may have many different types of notes that you want to display in different ways on your Flowershow website. In this tutorial we will cover how to add a new `Recipe` document type for our cooking blog as well as a custom `RecipeLayout` for that specific document type. Then we will create a home page for listing all the pages of this type.
+
+Let's start with creating a new `Recipe` document type.
+
+## Create a custom Contentlayer document type
+
+In order to create our new `Recipe` document type, we need to go inside `.flowershow/contentlayer.config.ts` and add instructions to tell Contentlayer how to parse:
+
+- `name`: namespace
+- `filePathPattern`: input files (Files within this path do not need to specify their `Type` to `Recipe`)
+- `fields`: meta data fields
+- `computedFields`: derived meta data fields
+
+```javascript
+//contentlayer.config.ts
+// ...
+const recipeFields = {
+  created: { type: "date", required: true },
+  authors: {
+    type: "list",
+    of: { type: "string" },
+  },
+  layout: { type: "string", default: "recipe" },
```

Comment:
But we're not suing any new fields here 😭. Take a look at this example recipe page: https://www.bbcgoodfood.com/recipes/classic-lasagne-0. We could have a lot of fancy stuff added to frontmatter that we could then use in the custom layout. We could even have a field for ingredients list so that we can display them in some box on the left or sth like that.